Update doc for search (#2341)
infiniflow/infinity: docs/guides/search_guide.md

Both RAG tokenization and fine-grained RAG tokenization are used in RAGFlow to ensure high recall.
:::

#### IK analyzer

> The IK analyzer is a bilingual tokenizer that supports Chinese (simplified and traditional) and English. It is a C++ adaptation of [IK Analyzer](https://github.com/infinilabs/analysis-ik) which is a very popular tokenizer for Chinese elasticsearch users.

Reviewer: ```suggestion
The IK analyzer is a bilingual tokenizer that supports Chinese (simplified and traditional) and English. It is a C++ adaptation of the [IK Analyzer](https://github.com/infinilabs/analysis-ik), widely used as a tokenizer by Chinese Elasticsearch users.
```

> Use `"ik"` to select the IK analyzer or `"ik-fine"` for fine-grained mode. `"ik"` works the same with `ik_smart` of `IK Analyzer` while `"ik-fine"` works the same with `ik_max_word` of `IK Analyzer`.

Reviewer: ```suggestion
Use `"ik"`/`"ik_smart"` to select the IK analyzer or `"ik-fine"`/`"ik_max_word"` for fine-grained mode.
```
